src/Application/GameConfig.h

```diff
@@ -4,13 +4,1474 @@
+#define GAME_FLAGS_1_DRAW_BLV_DEBUGS            0x08     // draw portal debug outlines
+#define GAME_FLAGS_1_40                         0x40     //
+#define GAME_FLAGS_1_FORCE_LEGACY_PROJECTION    0x80     // force camera / culling / projection mode : BLV
+#define GAME_FLAGS_1_400                        0x400    //
+#define GAME_FLAGS_1_800                        0x800    //
+#define GAME_FLAGS_1_TORCHLIGHT                 0x1000   // do Torchlight lightning
+#define GAME_FLAGS_1_40000_sw_alter_palettes    0x40000  // software render alter palettes
+#define GAME_FLAGS_1_HUGE_BLOODSPLATS           0x80000  // huge bloodsplats on bosses
+
+#define GAME_FLAGS_2_FORCE_REDRAW               0x01     // force redraw this frame
+#define GAME_FLAGS_2_SATURATE_LIGHTMAPS         0x02     // alter lightmaps saturation
+#define GAME_FLAGS_2_ALLOW_DYNAMIC_BRIGHTNESS   0x04     // allows dynamic brightness calculation on lightmaps
+#define GAME_FLAGS_2_ALTER_GRAVITY              0x08     // underwater effect, no gravity (for out15.odm - Shoals)
+#define GAME_FLAGS_2_TARGETING_MODE             0x10     // spell targeting, current cursor is "MICON2" (crosshair)
+#define GAME_FLAGS_2_DRAW_BLOODSPLATS           0x20
+
 using Engine_::CommandLine;
 
 namespace Application {
     class GameConfig {
      public:
-        explicit GameConfig(const std::string& comamnd_line);
+        explicit GameConfig(const std::string &command_line);
+        void DefaultConfiguration();
+        void LoadConfiguration();
+        void SaveConfiguration();
+
+        bool SetCustomBool(std::string section, std::string key, bool value);
+        bool SetCustomInteger(std::string section, std::string key, int value);
+        bool SetCustomFloat(std::string section, std::string key, float value);
+        bool SetCustomString(std::string section, std::string key, std::string value);
 
+        bool GetCustomBool(std::string section, std::string key, bool def_value);
+        int GetCustomInteger(std::string section, std::string key, int def_value);
+        float GetCustomFloat(std::string section, std::string key, float def_value);
+        std::string GetCustomString(std::string section, std::string key, std::string def_value);
+
+        const std::string config_file = "womm.ini";
         std::shared_ptr<CommandLine> command_line;
+
+        class Debug {
+         public:
+            const std::string section_name = "debug";
+
+            void Default();
+            void Load();
+            void Save();
+
+            inline bool DefaultAllMagic() {
+                return false;
+            }
+            inline void SetAllMagic(bool is_enabled) {
+                this->all_magic = is_enabled;
+            }
+            inline bool GetAllMagic() {
+                return this->all_magic;
+            }
+            inline void ToggleAllMagic() {
+                this->all_magic = !this->all_magic;
+            }
+
+            inline bool DefaultInfiniteFood() {
```

Comment:
Tbh I'm not even sure why we need all this boilerplate given that everything is writable.

Let's just use PODs?